chromium_src/chrome/common/icon_loader_auralinux.cc

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "chrome/common/icon_loader.h"
+
+#include "base/bind.h"
+#include "base/message_loop/message_loop.h"
+#include "base/nix/mime_util_xdg.h"
+#include "ui/views/linux_ui/linux_ui.h"
+
+// static
+IconGroupID IconLoader::ReadGroupIDFromFilepath(
+    const base::FilePath& filepath) {
+  return base::nix::GetFileMimeType(filepath);
+}
+
+// static
+bool IconLoader::IsIconMutableFromFilepath(const base::FilePath& path) {
+  return false;
+}
+
+void IconLoader::ReadIcon() {
+  int size_pixels = 0;
+  switch (icon_size_) {
+    case IconLoader::SMALL:
+      size_pixels = 16;
+      break;
+    case IconLoader::NORMAL:
+      size_pixels = 32;
+      break;
+    case IconLoader::LARGE:
+      size_pixels = 48;
+      break;
+    default:
+      NOTREACHED();
+  }
+
+  views::LinuxUI* ui = views::LinuxUI::instance();
+  if (ui) {
```

Comment:
This will always be null currently since we havn't set https://cs.chromium.org/chromium/src/ui/views/linux_ui/linux_ui.h?q=linuxui&sq=package:chromium&l=73 , hence no icon will be empty on linux. Have to fix it.